tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,269 @@
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        event.set()
+        for pod in pod_list:
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 3.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 3.1: Checked cluster has failures.")
+        LOGGER.info("Step 3: Performed soft-failure of %s control pods in loop and checked cluster "
+                    "has failures", self.repl_num - 1)
+
+        time.sleep(15)
+        event.clear()
+        LOGGER.info("Waiting for background threads to join")
+        thr_iam.join()
+        io_thread.join()
+        LOGGER.info("Step 4: Verifying responses from background processes")
+        LOGGER.info("Step 4.1: Verify background process for IAM user creation")
+        iam_resp = tuple()
+        while len(iam_resp) != 4:
+            iam_resp = iam_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not iam_resp:
+            assert_utils.assert_true(False, "Background process failed to do IAM user CRUD "
+                                            "operations")
+        failed = iam_resp[1]
+        user_del_failed = iam_resp[2]
+        created_users = iam_resp[3]
+        if failed:
+            assert_utils.assert_true(False, "No IAM user creation/deletion expected to fail before "
+                                            "or after soft-failure: {failed}")
+        if created_users:
+            for i_i in created_users:
+                self.s3_clean.update(i_i)
+        LOGGER.info("Updating dict for clean up with remaining users")
+        for i_i in user_del_failed:
+            self.s3_clean.update({i_i: del_users[i_i]})
+        LOGGER.info("Step 4.1: Verified background process for IAM user creation/deletion")
+
+        LOGGER.info("Step 4.2: Verify background process of IOs")
+        responses = {}
+        while len(responses) != 2:
+            responses = io_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses["pass_res"])
+        fail_logs = list(x[1] for x in responses["fail_res"])
+        resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
+        assert_utils.assert_false(len(resp[1]), f"Expected Pass, But Logs which contain failures:"
+                                                f" {resp[1]}")
+        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs, pass_logs=False)
```

Comment:
We expect all logs to be passed.
Need pass_logs = True , and can combine both the lists for verification.